src/backend/distributed/executor/intermediate_results.c

```diff
@@ -940,6 +943,19 @@ fetch_intermediate_results(PG_FUNCTION_ARGS)
 static uint64
 FetchRemoteIntermediateResult(MultiConnection *connection, char *resultId)
 {
+	char *localPath = QueryResultFileName(resultId);
+
+	struct stat fileStat;
+	int statOK = stat(localPath, &fileStat);
+	if (statOK == 0)
```

Comment:
My preference would be
```c
if (!stat(localPath, &fileStat))
```
or
```c
if (stat(localPath, &fileStat) == 0)
```

but obviously you see value in using a variable there. So, this is my suggestion:
```suggestion
	bool statOK = stat(localPath, &fileStat) == 0;
	if (statOK)
```

Reply:
Not sure about this suggestion. The current code seems super explicit about which return code means that the file exists.

`if (!stat(localPath, &fileStat)) ` would be very confusing because we actually go into this code path when `stat` succeeded.

Reply:
Then the name is not that good, because `statOK` means "if `stat` returned OK" to me at first, rather than "return value of `stat`". If you don't like my boolean suggestion (which is more explicit IMO, saying it means `stat` returned OK when the return value is 0), you can change the variable name to `statResult` or `statRet` or more explicitly `statReturnValue`.

Reply:
not sure, we seem to have used statOK in 2 other places in this file. I'd favour consistency

Reply:
We could still correct (or improve) the names at the other places, if consistency is favored. I see 5 other usages, it shouldn't be very difficult to replace them.